identity-composer-directive.js

```diff
@@ -34,38 +34,221 @@ function brIdentityComposer($rootScope, brCredentialLibraryService) {
+      scope.claimsPartiallyFulfillable = false;
+      for(var key in scope.selectedCredentials) {
+        var credential = scope.selectedCredentials[key];
+        var claims = scope.claimsForCredential(credential);
+        for(key in claims) {
+          var claim = claims[key];
+          if(claim in scope.requestedProperties) {
+            scope.requestedProperties[claim].fulfillable = true;
+            scope.claimsPartiallyFulfillable = true;
+          }
+        }
+      }
+      for(var key in scope.requestedProperties) {
+        var requestedProperty = scope.requestedProperties[key];
+        if(!requestedProperty.fulfillable) {
+          scope.page = 'unfulfillable'
+        }
+      }
     };
 
-    // TODO: remove once br-credential-thumbnail is available
-    scope.showCredential = function(option) {
-      scope.modal.show = true;
-      scope.modal.credential = option.credential;
-      scope.modal.groups = option.credentialGroups;
+    scope.isReplacing = function() {
+      for(var key in scope.selectedCredentials) {
+        var credential = scope.selectedCredentials[key];
+        if(credential.editing) {
+          return true;
+        }
+      }
+      return false;
+    }
+    
+    scope.back = function() {
+      if(scope.isReplacing()) {
+        for(var key in scope.selectedCredentials) {
+          var credential = scope.selectedCredentials[key];
+          credential.hidden = false;
+          credential.editing = false;
+        }
+      } else {
+        for(var key in scope.selectedCredentials) {
+          var credential = scope.selectedCredentials[key];
+          scope.page = 'front';
+        }
+      }
+    }
+
+    scope.clickItem = function(credential) {
+      // Mark the clicked credential editing, and hide all other credentials
+      scope.selectedPage = false;
+      credential.hidden = false;
+      credential.editing = true;
+      for(var key in scope.selectedCredentials) {
+        var selectedCredential = scope.selectedCredentials[key];
+        selectedCredential.hidden = true;
+      }
     };
 
+    scope.claimsForCredential = function(credential) {
+      // This only pulls in claims at the top level, and does not recover nested claims
+      var claims = [];
+      for(var key in credential.claim) {
+        if(key === 'id' || key === 'image') {
+          continue;
+        }
+        claims.push(key)
+      }
+      return claims;
+    };
+
+    scope.fulfillsExcessClaims = function(credential) {
+      var claims = scope.claimsForCredential(credential);
+
+      var requestedClaims = scope.requestedProperties;
+
+      var excessClaims = [];
+      for(var key in claims) {
+        var claim = claims[key];
+        if(!(claim in requestedClaims)) {
+          excessClaims.push(claim);
+        }
+      }
+      return excessClaims;
+    }
+
+    scope.htmlClaims = function() {
+      var html = '<h6>The site is requesting the following information:</h6>';
```

Comment:
I don't think it's possible -- bootstrap tooltips don't support newlines, so the only way to get a formatted tooltip is through html, but because of how the boostrap-angular tooltip gets called in the view, its either not possible or would be really complex to embed in the html there.